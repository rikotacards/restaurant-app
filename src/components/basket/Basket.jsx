import React from "react";
import "./basket.css";
import { Card } from "components/card/Card";
import { getTotalBasketPrice } from "../../util/getTotalBasketPrice";

const style = {
  position: "sticky",
  top: "100px"
};

export const Basket = props => {
  const { basketItems } = props;
const disableButton = basketItems.length === 0; 
  const basketLineItems = basketItems.map((item, index) => {
    const { quantity, name, price, options, selections } = item;
    return (
      <div className="basket-line-item" key={name+index}>
        <p>
          {quantity} x {name} HK${price} <br />({options}, {selections})
        </p>
      </div>
    );
  });

  const total = getTotalBasketPrice(basketItems);

  return (
    <Card style={style}>
      <div className="basket-header">Your Basket</div>
      {basketLineItems.length > 0 ? basketLineItems : "Your basket is empty"}
      {basketLineItems.length > 0 && <p>HK$ {total}</p>}
      <button disabled={disableButton} className="menu-item-button">Check out</button>
    </Card>
  );
};
