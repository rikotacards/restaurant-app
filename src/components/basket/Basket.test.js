import React from 'react'; 
import { Basket } from './Basket';
import renderer from 'react-test-renderer';

describe('Render Basket', () => {
    it('show Your basket is empty text', () => {
        const testDataEmpty = [];
    const tree = renderer
      .create(<Basket basketItems={testDataEmpty}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    }),

    it('should show one line item', () => {
        const testData = [{price:50}];
        const tree = renderer
          .create(<Basket basketItems={testData}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
})



 
  