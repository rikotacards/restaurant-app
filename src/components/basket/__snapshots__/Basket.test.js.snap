// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Basket should show one line item 1`] = `
<div
  className="card"
  style={
    Object {
      "position": "sticky",
      "top": "100px",
    }
  }
>
  <div
    className="basket-header"
  >
    Your Basket
  </div>
  <div
    className="basket-line-item"
  >
    <p>
       x 
       HK$
      50
       
      <br />
      (
      , 
      )
    </p>
  </div>
  <p>
    HK$ 
    50
  </p>
  <button
    className="menu-item-button"
  >
    Check out
  </button>
</div>
`;

exports[`Render Basket show Your basket is empty text 1`] = `
<div
  className="card"
  style={
    Object {
      "position": "sticky",
      "top": "100px",
    }
  }
>
  <div
    className="basket-header"
  >
    Your Basket
  </div>
  Your basket is empty
  <button
    className="menu-item-button"
  >
    Check out
  </button>
</div>
`;
